### DISCUSSION TITLE
The airflow db migration command does not work.

### DISCUSSION THREAD

**Discussion by jojaeng2** [apache/airflow#37441] [Q&A] [answered]
### Airflow version
2.8.0

### python version
3.11.5

I'm going to run the airflow db migration command to configure airflow charts in different environments, but I don't know why it doesn't work with only the output below

`airflow db migrate -v `

<img width="1877" alt="image" src="https://github.com/apache/airflow/assets/76645095/27923b70-7aa8-49e8-a16d-12d948fbd6e7">
I don't know why the hell the invalid command db is coming out.

`airflow check -v`

<img width="1920" alt="image" src="https://github.com/apache/airflow/assets/76645095/9cef5e96-f043-4d37-9323-b2a622f74d3d">

The DB connection is well established.
![image](https://github.com/apache/airflow/assets/76645095/202ee533-ffcd-4358-96a5-80ef61f89ee2)
Even if you output airflow info, it comes out well.

I really hope someone can help me. I can't fix it because I don't get any logs. Thank you.

**Comment by jojaeng2** [comment]
The `airflow db migrate --show-sql-only` command results in the following

![image](https://github.com/apache/airflow/assets/76645095/520e3542-375f-4e00-b006-5e34c6a72c6c)

**Comment by jojaeng2** [comment]
If I check the alembic_version in my metadata db, it's '10b52ebd31f7'. 

And this is consistent with the value of 2.8.0 Head for Airflow docs.

**Reply by potiuk** [reply, marked as answer]
Similar issue here: https://github.com/apache/airflow/discussions/18408

Somewhere in your DAGs you have a code that initializes argparse and fails because it does not understand `db` command. Unfortunatley argparse in this case will print "airlfow" as a culprit (because that was the original script), but in this case it's some other argparse intitailization that causes it.

Likely in some of your DAGs or settings you have some debug/test code that is not guarded with:

```python
if __name__ == '__main__':
     your test code here
```

And it gets executed during DAG parsin. Airflow's migrations will actually parse all the DAGs while running.

**Reply by jojaeng2** [reply]
Thank you very much, sir
I think the cause of the problem was that there was setup.py in AIRFLOW_DAG__HOME, and this setup.py used find_packages. But the error log said it was related to DB command, so I was spending time in strange places.
